File: presentation/samples/point.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
  int x, y;
} point;

point add(point a, point b)
{
  point result = { a.x + b.x, a.y + b.y };
  return result;
}

int main(int argc, char* argv[])
{
  point a = { 1, 2 };
  point* b = malloc(sizeof(point));
  b->x = 4;
  b->y = 3;
  point c = add(a, *b);
  printf("{.x = %d, .y = %d}\n", c.x, c.y);
  return 0;
}

File: presentation/samples/point.rs
struct Point {
    x: int,
    y: int
}

impl Point {
    fn add(self, other: Point) -> Point {
        Point { x: self.x + other.x,
                y: self.y + other.y }
    }
}

fn main() {
    let a = Point { x: 1, y: 2 };
    let b = ~Point { x: 4, y: 3 };
    println!("{:?}", a.add(*b));
}
